BLSQ/iaso
# WC2-226 Integrate WFP CIAM as am auth provider

Allow login via WFP auth server. see the README.md.

Will re-conciliate user on email address, it will automatically create an user account if there is none present (with the username as email). will send an email to warn a new account has been created.

It still missing a special page if the user has no permission at all, I'm on it but I'm already opening this PR

Related JIRA tickets : WC2-226  & WC2-228 & WC2-227

## Self proofreading checklist

- [na] Are my typescript files well typed
- [] New translations have been added or updated if new strings have been introduced in the frontend
- [na] My migrations file are included
- [NO] Are there enough tests

## Changes

- upgraded django-allauth to a version that support the PKCE challenge
- Move the whole code in a separate django app.
- Removed the old code
- see `plugins/wfp_auth/README.md`

## How to test

To test in local:

set the environment variable
-       WFP_AUTH_CLIENT_ID: Check in one password
-       WFP_AUTH_ACCOUNT: Which Iaso account user should be associated to
-       WFP_EMAIL_RECIPIENTS_NEW_ACCOUNT: email that should receive the notification for a new user.

Try login via the CIAM button (it's only added if the above has been configured). Create an account on the WFP server, validate the e-mail. You can try login before the email is not validated, it should fail.


## Print screen / video


https://github.com/BLSQ/iaso/assets/82500/caa047b2-b243-46d6-9473-ca000a6c2d0e


Note that the front end button has not been translated so far.  
The email haven't been translated either but I'm not sure this make sense
